src/llvm_backend_stmt.cpp

```diff
@@ -2917,6 +2917,22 @@ gb_internal void lb_emit_defer_stmts(lbProcedure *p, lbDeferExitKind kind, lbBlo
 	}
 	defer (p->branch_location_pos = prev_token_pos);
 
+	// TODO(lucas): In LLVM 21 use the 'use-after-scope' asan option which does this for us.
+	#if LLVM_VERSION_MAJOR < 21
+		if (kind == lbDeferExit_Return) {
+			for_array(i, p->asan_stack_locals) {
+				lbValue local = p->asan_stack_locals[i];
+
+				auto args = array_make<lbValue>(temporary_allocator(), 2);
+				args[0] = lb_emit_conv(p, local, t_rawptr);
+				args[1] = lb_const_int(p->module, t_int, type_size_of(local.type->Pointer.elem));
+				lb_emit_runtime_call(p, "__asan_unpoison_memory_region", args);
+			}
+		}
+	#else
+		#error "Need to implement LLVM 21 'use-after-scope' asan option"
```

Comment:
I would do `compiler_error(...)` if an `p->asan_stack_locals` exists rather than prevent people from compiling the compiler.

Reply:
Hmm I think if you want that it is better if I just have no error and just stay vigilant for the llvm 21 release